core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -49,11 +50,13 @@ class Project(
   private val searchService = new SearchService(config, resolver)
   private val sourceWatcher = new SourceWatcher(config, resolver :: Nil)
   private val reTypecheck = new FileChangeListener {
-    def reTypeCheck(): Unit = self ! AskReTypecheck
-    def fileAdded(f: FileObject): Unit = reTypeCheck()
-    def fileChanged(f: FileObject): Unit = reTypeCheck()
-    def fileRemoved(f: FileObject): Unit = reTypeCheck()
-    override def baseReCreated(f: FileObject): Unit = reTypeCheck()
+    private val reTypeCheck = Debouncer("AskReTypecheck", system.scheduler, 5.seconds, 20.seconds) { () =>
```

Comment:
This is why I did the maxDelay of 20.seconds; it'll get called, eventually. But, we can totally make them configurable before. It was hard coded before.